cmd/guacone/cmd/s3.go

```diff
@@ -0,0 +1,154 @@
+//
+// Copyright 2023 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	csub_client "github.com/guacsec/guac/pkg/collectsub/client"
+	"github.com/guacsec/guac/pkg/handler/collector"
+	"github.com/guacsec/guac/pkg/handler/collector/s3"
+	"github.com/guacsec/guac/pkg/handler/processor"
+	"github.com/guacsec/guac/pkg/ingestor"
+	"github.com/guacsec/guac/pkg/logging"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"golang.org/x/sync/errgroup"
+)
+
+// s3Options flags for configuring the command
+type s3Options struct {
+	s3hostname string // hostname of the s3 provider
+	s3port     string // port of the s3 provider
+	region     string // AWS region, for s3/sqs configuration (defaults to us-east-1)
+	queues     string // comma-separated list of queues/topics
+	mp         string // message provider name (sqs or kafka, will default to kafka)
+	mpHostname string // hostname for the message provider
+	mpPort     string // port for the message provider
+}
+
+var s3Opts s3Options
+
+var s3Cmd = &cobra.Command{
+	Use:   "s3 [flags]",
+	Short: "listens to kafka/sqs s3 events to download documents and add them to the GUAC graph",
+	Args:  cobra.MinimumNArgs(0),
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx := logging.WithLogger(context.Background())
+		logger := logging.FromContext(ctx)
+
+		s3Opts, err := validateS3Opts(s3Opts.s3hostname, s3Opts.s3port, s3Opts.mp, s3Opts.mpHostname, s3Opts.mpPort, s3Opts.queues, s3Opts.region)
+		if err != nil {
+			os.Exit(1)
+		}
+
+		s3Collector, err := s3.NewS3Collector(s3.S3CollectorConfig{
+			S3Host:              s3Opts.s3hostname,
+			S3Port:              s3Opts.s3port,
+			MessageProvider:     s3Opts.mp,
+			MessageProviderHost: s3Opts.mpHostname,
+			MessageProviderPort: s3Opts.mpPort,
+			Queues:              s3Opts.queues,
+			Region:              s3Opts.region,
+		})
+		if err != nil {
+			logger.Errorf("unable to create s3 collector: %v", err)
+			os.Exit(1)
+		}
+
+		err = collector.RegisterDocumentCollector(s3Collector, s3.S3CollectorType)
+		if err != nil {
```

Comment:
use `if err := collector.RegisterDocumentCollector(s3Collector, s3.S3CollectorType); err != nil`
Whenever there is just err as the return value.

This way you don't have to keep track if err is defined or not and use `=` vs `:=` based on that.